### PR TITLE
[doc] documentation for the general architecture

### PR DESCRIPTION
**Pull request by gautierhattenberger**

### CODE REVIEW

File: doc/sphinx/source/developer_guide/system_overview.rst [hunk 1]
```diff
@@ -4,40 +4,105 @@
...
-.. image:: general_arch.png
+In the case of the Paparazzi UAV system, the ground segment is composed by a collection of tools connected by a middleware, the communication is based on the `PPRZLINK <https://pprzlink.readthedocs.io/en/latest/>`_ library and the airborne part supports several firmwares, including fixed-wing planes, rotorcrafts, hybrids and rovers
+
+.. image:: PPRZ_System_architecture.png
 
 Ground architecture
 -------------------
 
-The Gound Control Station (GCS) constits of multiples programs, communicating with each other by the software bus `Ivy <https://www.eei.cena.fr/products/ivy/>`_.
+The Gound Control Station (GCS) constits of multiples programs, communicating with each other by the software bus `Ivy <https://www.eei.cena.fr/products/ivy/>`_, as seen on the general architecture diagram above.
```
Comment (Fabien-B) [inline doc/sphinx/source/developer_guide/system_overview.rst:19]:
```suggestion
The Ground Control Station (GCS) consists of multiples programs, communicating with each other by the software bus `Ivy <https://www.eei.cena.fr/products/ivy/>`_, as seen on the general architecture diagram above.
```
and may replace the Ivy link for https://ivybus.gitlab.io/

File: doc/sphinx/source/developer_guide/system_overview.rst [hunk 2]
```diff
@@ -4,40 +4,105 @@
...
 
     Ivy is a simple protocol and a set of open-source (LGPL) libraries and programs that allows applications to broadcast information through text messages, with a publisher / subscriber mechanism based on regular expressions.
 
-.. image:: agents_arch.png
-
 The core Paparazzi programs are Link, which handle the communication with the drones, the Server, which maintain the state of all aircrafts, and the GCS, the user facing application to control the drones.
 
 Many other tools have been developed for various use cases, they are available in the *Tools* menu of the paparazzi center. But more interesting: you can write your own tools that will interact with Paparazzi via the Ivy bus.
 
+The ground segment useually runs on a single laptop, but thanks to its distributed nature, it is also possible to split the tools on different computers connected to the same network or VPN.
```
Comment (Fabien-B) [inline doc/sphinx/source/developer_guide/system_overview.rst:29]:
```suggestion
The ground segment usually runs on a single laptop, but thanks to its distributed nature, it is also possible to split the tools on different computers connected to the same network or VPN.
```

File: doc/sphinx/source/developer_guide/system_overview.rst [hunk 3]
```diff
@@ -4,40 +4,105 @@
...
+    or quaternion in the case of attitude representation), while they can be retrieved from any other components within AP process and in any format as well.
+    It means that if a format transformation is required (e.g. from quaternion to Euler), the conversion is performed on the fly, only once,
+    until a new update is available on the state. The available data in this interface are the position, the velocity, the acceleration,
+    the orientation (attitude), the rotation speed, the air and wind speed.
 
-For the rotorcraft, there is only one process called *Main*.
+  - The state is then used by the control loops for navigation, guidance and stabilization. The final commands array is stored in a normalized control vector.
+    These commands are ultimately applied to the physical actuators with a mixing defined in an airframe configuration file.
+
+  - Some other services are present, including the communication services allow to send and receive messages from the ground, from an other aircraft
```
Comment (Fabien-B) [inline doc/sphinx/source/developer_guide/system_overview.rst:55]:
```suggestion
  - Some other services are present, including the communication services that allow to send and receive messages from the ground, from an other aircraft
```

File: doc/sphinx/source/developer_guide/system_overview.rst [hunk 4]
```diff
@@ -4,40 +4,105 @@
...
-.. warning::
-    
-    TODO: Explain airborne architecture
+The aircraft is configured by various XML configuration files: *airframe*, *flight_plan*, *radio* and *telemetry*. theses will be covered in the :doc:`../user_guide/index_user_guide` section.
+
+Airborne autopilot timing
+-------------------------
 
-.. image:: airborne_arch.png
+The picture below explain the temporal sequence of the autopilot. Each module is declared with a *task* (or nothing for default group) that is called in a predefined order.
```
Comment (Fabien-B) [inline doc/sphinx/source/developer_guide/system_overview.rst:74]: Not very clear, maybe that ?
```suggestion
The picture below explain the temporal sequence of the autopilot. Modules are grouped in *tasks* (or nothing for default group) that are called in a predefined order.
```

**Review by Fabien-B — CHANGES_REQUESTED** [review]

File: doc/sphinx/source/developer_guide/system_overview.rst [hunk 5]
```diff
@@ -16,7 +16,7 @@ In the case of the Paparazzi UAV system, the ground segment is composed by a col
 Ground architecture
 -------------------
 
-The Gound Control Station (GCS) constits of multiples programs, communicating with each other by the software bus `Ivy <https://www.eei.cena.fr/products/ivy/>`_, as seen on the general architecture diagram above.
+The Ground Control Station (GCS) consits of multiples programs, communicating with each other by the software bus `Ivy <https://www.eei.cena.fr/products/ivy/>`_ (see the `documentation <https://ivybus.gitlab.io/>`_ for details), as seen on the general architecture diagram above.
```
Comment (fvantienen) [inline doc/sphinx/source/developer_guide/system_overview.rst:19]: consits -> consists

**Review by Fabien-B — APPROVED** [review]

**Pull request merged**